components/AutoComplete/AutoComplete.jsx

```diff
@@ -0,0 +1,356 @@
+      setShowSuggestions(false);
+      wrapperRef.current.children[1].focus();
+    }
+  }, [cursor, enterPress]);
+
+  /* istanbul ignore next */
+  useEffect(() => {
+    if (showSuggestions && tabPress) {
+      setUserTypedValue(filterSuggestions[cursor]);
+      setShowSuggestions(false);
+    } else if (tabPress) {
+      setShowSuggestions(false);
+    }
+  }, [tabPress]);
+
+  useEffect(() => {
+    window.addEventListener('click', handleClickOutside);
+    window.addEventListener('keydown', handleEscPress);
+    return () => {
+      window.removeEventListener('click', handleClickOutside);
+      window.removeEventListener('keydown', handleEscPress);
+    };
+  }, []);
+
+  return (
+    <Wrapper ref={wrapperRef}>
+      <InputLabel htmlFor={id} error={error}>
+        {label}
+      </InputLabel>
+      <InputText
+        id={id}
+        name={name}
+        type="text"
+        error={error}
+        disabled={disabled}
+        placeholder={placeholder}
+        autoComplete="off"
+        role="combobox"
+        aria-owns="autocompleteOptions"
+        aria-autocomplete="both"
+        aria-expanded={showSuggestions}
+        value={userTypedValue}
+        onClick={() => handleInputClick()}
+        onChange={e => handleChange(e)}
+      />
+      {userTypedValue && !error && !disabled && (
+        <InputIcon
+          theme={theme}
+          name="clear"
+          description="limpar valor"
+          onClick={() => handleClearValue()}
+        />
+      )}
+      {generateSuggestions()}
+      {helperText && <HelperText>{helperText}</HelperText>}
+      {error && <InputErrorMessage theme={theme}>{error}</InputErrorMessage>}
+      {error && <InputErrorIcon description={error} theme={theme} />}
+      <PoliteStatus role="status" aria-atomic="true" aria-live="polite">
+        {generateAssistiveDescript()}
+      </PoliteStatus>
+    </Wrapper>
+  );
+};
+
+AutoComplete.propTypes = {
+  theme: PropTypes.shape({
+    colors: PropTypes.object,
+    spacing: PropTypes.object,
+  }),
+  /** A list of string or objects with the values to show in component */
+  suggestions: PropTypes.arrayOf(PropTypes.string).isRequired,
+  id: PropTypes.string,
+  /** Displays an error message and changes border color to error color */
+  error: PropTypes.string,
+  disabled: PropTypes.bool,
+  /** Displays a label text that describes the field */
+  label: PropTypes.string,
+  name: PropTypes.string,
+  placeholder: PropTypes.string,
+  /** Callback function to receive user typed value and user selected value */
+  selectedItem: PropTypes.func,
+  /** Displays a helper text below the component */
+  helperText: PropTypes.string,
+};
+
+AutoComplete.defaultProps = {
+  theme: {
+    spacing,
+    colors,
+  },
+  id: '',
+  name: '',
+  label: '',
+  helperText: '',
+  error: '',
+  disabled: false,
+  placeholder: 'Select an option',
+  selectedItem: () => {},
+};
+
+export default AutoComplete;
```

Comment:
is coverage normal?

![2022-01-04_17-03](https://user-images.githubusercontent.com/190265/148117084-a3ffa94e-dc90-4673-83b1-47d6b7970d45.png)